src/main/python/vantiqsdk.py

```diff
@@ -1458,32 +1462,8 @@ async def connect(self, do_pings: bool = True):
 
     async def subscribe(self, path: str, params: dict, callback: Callable[[str, dict], Awaitable[None]],
                         target_namespace: Union[str, None] = None) -> VantiqResponse:
-        if self.connected:
-            if path in self.subscriptions.keys():
-                vr = VantiqResponse(False, 400, None)
-                ve = VantiqError('io.vantiq.python.subscribed',
-                                 'A subscription for path {0} already exists.',
-                                 [path])
-                vr.errors = [ve]
-                return vr
-            self.callbacks[path] = callback
-            self.subscriptions[path] = False  # Will be set to true when the server responds
-            if params is None:
-                params = {}
-            params['requestId'] = path
-            sub_msg = {'op': 'subscribe',
-                       'resourceName': 'events',
-                       'resourceId': path,
-                       'parameters': params}
-            target_namespace = target_namespace or self.parent.get_target_namespace()
-            if target_namespace is not None:
-                sub_msg['targetNamespace'] = target_namespace
-            await self.connection.send(json.dumps(sub_msg))
-            self._vlog.debug('Subscription request sent.')
-
-            vr = VantiqResponse(True, 204, None)
-            return vr
-        else:
+        # If we aren't connected, complain
```

Comment:
Flipped this to avoid a long main body and a short else (I don't like the way those read).

Reply:
probably a leftover habit (which doesn't apply much in interpreted cases anyway) of keeping the most common cases as straight-line code.  Avoids cache flushes.

In new days, doesn't really matter any more.  Sorry.